src/engine/GraphStoreProtocol.h

```diff
@@ -0,0 +1,134 @@
+    };
+    updateClause::GraphUpdate up{
+        ad_utility::transform(triples, transformTurtleTriple), {}};
+    res._clause = parsedQuery::UpdateClause{up};
+    return res;
+  }
+  FRIEND_TEST(GraphStoreProtocolTest, transformPost);
+
+  static ParsedQuery transformGet(const GraphOrDefault& graph) {
+    ParsedQuery res;
+    res._clause = parsedQuery::ConstructClause(
+        {{Variable("?s"), Variable("?p"), Variable("?o")}});
+    res._rootGraphPattern = {};
+    parsedQuery::GraphPattern selectSPO;
+    selectSPO._graphPatterns.emplace_back(parsedQuery::BasicGraphPattern{
+        {SparqlTriple(Variable("?s"), "?p", Variable("?o"))}});
+    if (std::holds_alternative<ad_utility::triple_component::Iri>(graph)) {
+      parsedQuery::GroupGraphPattern selectSPOWithGraph{
+          std::move(selectSPO),
+          std::get<ad_utility::triple_component::Iri>(graph)};
+      res._rootGraphPattern._graphPatterns.emplace_back(
+          std::move(selectSPOWithGraph));
+    } else {
+      AD_CORRECTNESS_CHECK(std::holds_alternative<DEFAULT>(graph));
+      res._rootGraphPattern = std::move(selectSPO);
+    }
+    return res;
+  }
+  FRIEND_TEST(GraphStoreProtocolTest, transformGet);
+
+ public:
+  // Every Graph Store Protocol requests has equivalent SPARQL Query or Update.
+  // Transform the Graph Store Protocol request into it's equivalent Query or
+  // Update.
+  static ParsedQuery transformGraphStoreProtocol(
+      const ad_utility::httpUtils::HttpRequest auto& rawRequest) {
+    ad_utility::url_parser::ParsedUrl parsedUrl =
+        ad_utility::url_parser::parseRequestTarget(rawRequest.target());
+    GraphOrDefault graph = extractTargetGraph(parsedUrl.parameters_);
+
+    using enum boost::beast::http::verb;
+    auto method = rawRequest.method();
+    if (method == get) {
+      return transformGet(graph);
+    } else if (method == put) {
+      throw std::runtime_error(
+          "PUT in the SPARQL Graph Store HTTP Protocol is not yet implemented "
+          "in QLever.");
+    } else if (method == delete_) {
+      throw std::runtime_error(
+          "DELETE in the SPARQL Graph Store HTTP Protocol is not yet "
+          "implemented in QLever.");
+    } else if (method == post) {
+      return transformPost(rawRequest, graph);
+    } else if (method == head) {
+      throw std::runtime_error(
+          "HEAD in the SPARQL Graph Store HTTP Protocol is not yet implemented "
+          "in QLever.");
+    } else if (method == patch) {
+      throw std::runtime_error(
+          "PATCH in the SPARQL Graph Store HTTP Protocol is not yet "
+          "implemented in QLever.");
+    } else {
+      throw std::runtime_error(
+          absl::StrCat("Unsupported HTTP method \"",
+                       std::string_view{rawRequest.method_string()},
+                       "\" for the SPARQL Graph Store HTTP Protocol."));
+    }
+  }
+
+ private:
+  // Extract the graph to be acted upon using `Indirect Graph
+  // Identification`.
```

Comment:
Please elaborate what `Indirect Graph Identification` is, or where this term is defined.